### PR TITLE
React playground, react patterns

### PR DESCRIPTION
**Pull request by dperpeet**
replaces #4255

### CODE REVIEW

**Comment by dperpeet** [comment]
@mvollmer This adds extra action buttons to the dialog and the "danger" style
It also changes the usage slightly: Actions are now a list (not just primary action) and there's no callback anymore to the action click, use promise.notify instead

**Comment by dperpeet** [comment]
I want to add the listing pattern as a separate pull request, otherwise this will grow too complex

File: pkg/tuned/dialog.js [hunk 1]
```diff
@@ -122,46 +122,34 @@ define([
...
-                            dfd.reject(ex);
+                            return null;
                         });
                 } else {
-                    tuned.call('/Tuned', 'com.redhat.tuned.control', 'switch_profile', [ profile ])
+                    return tuned.call('/Tuned', 'com.redhat.tuned.control', 'switch_profile', [ profile ])
                         .then(function(results) {
                             if (!results[0][0]) {
-                                dfd.reject(results[0][1] || _("Failed to switch profile"));
+                                return cockpit.reject(results[0][1] || _("Failed to switch profile"));
```
Comment (stefwalter) [inline pkg/tuned/dialog.js]: Unfortunately cockpit.reject() doesn't exist yet. I'll make a pull request that adds it.

Comment (dperpeet) [inline pkg/tuned/dialog.js]: > Unfortunately cockpit.reject() doesn't exist yet. I'll make a pull request that adds it.

added in https://github.com/cockpit-project/cockpit/commit/153af793ccb0066b1815842401e41251db6acab8

**Pull request closed without merging**